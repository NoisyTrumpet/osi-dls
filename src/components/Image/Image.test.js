import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Image } from './';

describe('<Image />', () => {
  test('props should render down to <img/>', () => {
    const imageComponent = <Image src="test-src" alt="test-alt" className="test-class" />;
    const wrapper = shallow(imageComponent);
    expect(wrapper.getElement()).toBeTruthy();
    expect(wrapper.find('picture').length).toBe(1);
    expect(wrapper.find('img').prop('src')).toEqual('test-src');
    expect(wrapper.find('img').prop('alt')).toEqual('test-alt');
    expect(wrapper.find('img').prop('className')).toEqual('test-class');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('soucre tags should render', () => {
    const imageComponent = (
      <Image src="test-src" alt="test-alt" className="test-class">
        <Image.Source />
        <Image.Source />
      </Image>
    );
    const wrapper = shallow(imageComponent);
    expect(wrapper.find('picture').length).toBe(1);
    expect(wrapper.find(Image.Source).length).toBe(2);
    expect(wrapper.find('img').prop('src')).toEqual('test-src');
    expect(wrapper.find('img').prop('alt')).toEqual('test-alt');
    expect(wrapper.find('img').prop('className')).toEqual('test-class');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('picture attributes should render', () => {
    const imageComponent = (
      <Image pictureClass="pic-class" ariaHidden="true" src="test-src" alt="test-alt" className="test-class">
        <Image.Source srcSet="test1.jpg" dataSrcSet="data1" media="(min-width: 800px)" />
      </Image>
    );
    const wrapper = shallow(imageComponent);
    expect(wrapper.find('img').length).toBe(1);
    expect(wrapper.find('img').prop('aria-hidden')).toEqual('true');
    expect(wrapper.find('picture').length).toBe(1);
    expect(wrapper.find('picture').prop('className')).toEqual('pic-class');
    expect(wrapper.find(Image.Source).length).toBe(1);
    expect(wrapper.find(Image.Source).prop('srcSet')).toEqual('test1.jpg');
    expect(wrapper.find(Image.Source).prop('dataSrcSet')).toEqual('data1');
    expect(wrapper.find(Image.Source).prop('media')).toEqual('(min-width: 800px)');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
