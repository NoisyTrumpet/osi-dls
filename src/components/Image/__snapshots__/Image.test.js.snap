// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> picture attributes should render 1`] = `
<LazyLoad
  className="lazy-image-container"
  debounce={false}
  elementType="div"
  offset={0}
  offsetBottom={0}
  offsetHorizontal={128}
  offsetLeft={0}
  offsetRight={0}
  offsetTop={0}
  offsetVertical={128}
  throttle={250}
>
  <picture
    className="pic-class"
  >
    <ImageSource
      dataSrcSet="data1"
      media="(min-width: 800px)"
      srcSet="test1.jpg"
    />
    <img
      alt="test-alt"
      aria-hidden="true"
      className="test-class"
      src="test-src"
    />
  </picture>
</LazyLoad>
`;

exports[`<Image /> props should render down to <img/> 1`] = `
<LazyLoad
  className="lazy-image-container"
  debounce={false}
  elementType="div"
  offset={0}
  offsetBottom={0}
  offsetHorizontal={128}
  offsetLeft={0}
  offsetRight={0}
  offsetTop={0}
  offsetVertical={128}
  throttle={250}
>
  <picture>
    <img
      alt="test-alt"
      className="test-class"
      src="test-src"
    />
  </picture>
</LazyLoad>
`;

exports[`<Image /> soucre tags should render 1`] = `
<LazyLoad
  className="lazy-image-container"
  debounce={false}
  elementType="div"
  offset={0}
  offsetBottom={0}
  offsetHorizontal={128}
  offsetLeft={0}
  offsetRight={0}
  offsetTop={0}
  offsetVertical={128}
  throttle={250}
>
  <picture>
    <ImageSource />
    <ImageSource />
    <img
      alt="test-alt"
      className="test-class"
      src="test-src"
    />
  </picture>
</LazyLoad>
`;
